packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -680,6 +679,21 @@ const PipelineWrapper: React.FC<IProps> = ({
       }
     ],
     rightBar: [
+      {
+        action: '',
+        label: `Runtime: ${pipelineRuntime?.display_name ?? 'Generic'}`,
+        incLabelWithIcon: 'before',
+        enable: false,
+        kind: 'tertiary',
+        // TODO: use getRuntimeIcon
+        iconEnabled: IconUtil.encode(
+          pipelineRuntime?.name === 'kfp'
+            ? kubeflowIcon
+            : pipelineRuntime?.name === 'airflow'
+            ? airflowIcon
```

Comment:
Hmm - with my BYO-Runtime hat on, should the runtime's icon be a property of the runtime metadata?  It could probably be [defined as a constant in the schema](https://github.com/elyra-ai/elyra/blob/master/elyra/metadata/schemas/metadata-test.json#L68) with `pipelineIcon` as the fallback in the UI.

The current implementation looks great, btw!  Just thinking out loud about where we want to be.

Reply:
We've had some discussion on how the runtime icons should be loaded and due to the icon implementation in JupyterLab it's going to be loaded by the frontend for now (and maybe in the future it can be added to the backend as an option). I already added a function to #1620 to handle getting an icon for a runtime, but since it hasn't been merged yet, I added a TODO comment to switch to using the implementation for a `getRuntimeIcon` function when possible.

Reply:
Excellent - thanks Martha!